frontend/appflowy_flutter/lib/plugins/database/widgets/row/cells/date_cell/date_editor.dart

```diff
@@ -54,31 +98,146 @@ class _DateCellEditor extends State<DateCellEditor> {
+                .add(DateCellEditorEvent.setTime(timeStr)),
+            onEndTimeSubmitted: (timeStr) => context
+                .read<DateCellEditorBloc>()
+                .add(DateCellEditorEvent.setEndTime(timeStr)),
+            onDaySelected: (selectedDay, _) => context
+                .read<DateCellEditorBloc>()
+                .add(DateCellEditorEvent.selectDay(selectedDay)),
+            onRangeSelected: (start, end, _) => context
+                .read<DateCellEditorBloc>()
+                .add(DateCellEditorEvent.selectDateRange(start, end)),
           );
         },
       ),
     );
   }
+
+  void _removeReminder(ReminderBloc bloc, String? reminderId) {
+    if (reminderId != null) {
+      final reminder =
+          bloc.state.reminders.firstWhereOrNull((r) => r.id == reminderId);
+
+      if (reminder != null) {
+        bloc.add(ReminderEvent.remove(reminder: reminder));
+      }
+    }
+  }
+
+  void _updateReminderScheduledAt(
+    ReminderBloc bloc,
+    String reminderId,
+    DateTime scheduledAt,
+  ) =>
+      bloc.add(
+        ReminderEvent.update(
+          ReminderUpdate(
+            id: reminderId,
+            scheduledAt: scheduledAt.subtract(_reminderOption.time),
+          ),
+        ),
+      );
+
+  void _updateReminderOption(
+    ReminderOption newOption, {
+    required DateCellEditorBloc cellBloc,
+    required ReminderBloc reminderBloc,
+    required String rowId,
+  }) {
+    final dateOfEvent = cellBloc.state.dateTime;
+    if (dateOfEvent == null) {
+      return;
+    }
+
+    final reminderId = cellBloc.state.reminderId;
+    if ((reminderId != null || (reminderId?.isEmpty ?? false)) &&
+        newOption == ReminderOption.none) {
+      // Remove reminder if there is a reminder
+      final reminder = reminderBloc.state.reminders
+          .firstWhereOrNull((r) => r.id == reminderId);
+
+      if (reminder != null) {
+        reminderBloc.add(ReminderEvent.remove(reminder: reminder));
+      }
+
+      // Remove reminderId in database
+      return cellBloc.add(const DateCellEditorEvent.removeReminder());
+    } else if ((reminderId == null || reminderId.isEmpty) &&
+        newOption != ReminderOption.none) {
+      // Add reminder
+      final reminderId = nanoid();
+      final scheduledAtDate = dateOfEvent.subtract(newOption.time);
+      reminderBloc.add(
+        ReminderEvent.add(
+          reminder: ReminderPB(
+            id: reminderId,
+            objectId: widget.cellController.viewId,
+            title: LocaleKeys.reminderNotification_title.tr(),
+            message: LocaleKeys.reminderNotification_message.tr(),
+            scheduledAt: Int64(scheduledAtDate.millisecondsSinceEpoch ~/ 1000),
+            isAck: dateOfEvent.isBefore(DateTime.now()),
+            meta: {ReminderMetaKeys.rowId: rowId},
+          ),
+        ),
+      );
+
+      // Update reminderId in database
+      return cellBloc.add(
+        DateCellEditorEvent.setReminder(reminderId: reminderId),
```

Comment:
Idk... I don't think this logic belongs here.

Reply:
I'm unsure where else to put it, as it is deciding where to send events, to which BLoC. The BLoCs cannot communicate independantly.

And even if they could, they are not 1:1, eg. adding a reminder doesn't mean you add it to a Date Cell.

Reply:
I'll think about it, might be possible to do something with a Cubit.

Reply:
Got it